### PR TITLE
Issue #DP-1414 Fix:Adopted textbooks should not be editable

### PR DESCRIPTION
**Pull request by prashantsingh03**

### CODE REVIEW

File: src/app/client/src/app/modules/workspace/components/all-content/all-content.component.ts [hunk 1]
```diff
@@ -432,7 +432,15 @@ export class AllContentComponent extends WorkSpace implements OnInit, AfterViewI
   }
 
   contentClick(content) {
-    if (_.size(content.lockInfo)) {
+      if(content.originData){
+        var neww = JSON.parse(content.originData);
+        if((neww.copyType === 'shallow')){
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/workspace/components/all-content/all-content.component.ts:437]: why two brackets? @prashantsingh03 after if?

**Review by twsvaishali — APPROVED** [review]

**Review by twsvaishali — APPROVED** [review]
Approved

File: src/app/client/src/app/modules/workspace/components/all-content/all-content.component.ts [hunk 2]
```diff
@@ -432,7 +432,15 @@ export class AllContentComponent extends WorkSpace implements OnInit, AfterViewI
   }
 
   contentClick(content) {
-    if (_.size(content.lockInfo)) {
+      if (content.originData) {
+        const neww = JSON.parse(content.originData);
```
Comment (pallakartheekreddy) [inline src/app/client/src/app/modules/workspace/components/all-content/all-content.component.ts:436]: @prashantsingh03 please use proper variable names like `originData` insteadOf neww

File: src/app/client/src/app/modules/workspace/components/published/published.component.ts [hunk 3]
```diff
@@ -310,7 +310,15 @@ export class PublishedComponent extends WorkSpace implements OnInit, AfterViewIn
   */
   contentClick(param, content) {
     this.contentMimeType = content.metaData.mimeType;
-    if (param.action.eventName === 'delete') {
+    if (typeof param.data.originData !== "undefined") {
```
Comment (pallakartheekreddy) [inline src/app/client/src/app/modules/workspace/components/published/published.component.ts:313]: @prashantsingh03 same here

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**